app/scripts/modules/core/src/function/FunctionGroupings.tsx

```diff
@@ -0,0 +1,41 @@
+import * as React from 'react';
+import { Application } from '@spinnaker/core';
+import { IFunctionGroup } from 'core/domain';
+import { FunctionPod } from './FunctionPod';
+
+export interface IFunctionGroupingsProps {
+  app: Application;
+  groups: IFunctionGroup[];
+}
+export interface IFunctionGroupingsState {
+  groups: IFunctionGroup[];
+}
+export class FunctionGroupings extends React.Component<IFunctionGroupingsProps, IFunctionGroupingsState> {
+  constructor(props: IFunctionGroupingsProps) {
+    super(props);
+
+    const { groups } = props;
+    this.state = {
```

Comment:
You can just directly use the props instead of assigning `groups` to the state. Especially since you are never mutating the groups within the component. It will re-render anyway if the groups in props are updated.

Reply:
Pushed the change :)